swap_meet/vendor.py

```diff
@@ -1,2 +1,62 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self,inventory=None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+    
+    def add(self,item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self,item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+    
+    def get_by_id(self,id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.inventory.remove(my_item)
+            other_vendor.inventory.append(my_item)
+            other_vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            return True
+        return False
+  
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        my_first_item = self.inventory[0]
+        their_first_item = other_vendor.inventory[0]
+        self.inventory[0] = their_first_item
+        other_vendor.inventory[0] = my_first_item
+        return True
```

Comment:
These two functions are very similar! If you get a chance to refactor, I wonder if there is a way to rewrite them in a way where we could actually use one to implement the other! A couple things to think about here:

1. In our instructions, it doesn't explicitly state where the swapped items in `swap_items` need to be placed. Your method adds them to the end, but I wonder if there is a way to truly swap them by index?

2. In that same vein of swapping by index, there is a python list function called `index()` that returns the index of a specific item. It does have a O(n) runtime, but so does the remove() function...

This is not a required change to make, but it is something to think about if you decide to do any refactoring!